test/service_manager_test.dart

```diff
@@ -108,6 +108,101 @@ void main() {
       await _verifyExtensionStateInServiceManager(extensionName, true, 0.5);
     });
 
+    test('callService', () async {
+      dynamic _e;
+
+      final registeredService =
+          serviceManager.methodsForService['reloadSources'] ?? const [];
+      expect(registeredService, isNotEmpty);
+
+      try {
+        await serviceManager.callService('reloadSources',
+            isolateId: serviceManager.isolateManager.selectedIsolate.id);
+      } catch (e) {
+        _e = e;
+      }
+
+      expect(_e, isNull);
+    });
+
+    test('callService throws exception', () async {
+      // Service with less than 1 registration.
+      Exception exception;
+      Exception expectedException =
+          Exception('Expected one registered service for fakeMethod'
+              ' but found 0');
+
+      try {
+        await serviceManager.callService('fakeMethod');
+      } catch (e) {
+        exception = e;
+      }
+
+      expect(exception, isNotNull);
```

Comment:
package:test can make this logic a little cleaner to write.
For example you should be able to write
expect(service.manager.callService('fakeMethod'), throwsA(typeTypeItThrows));
https://pub.dartlang.org/documentation/test/latest/index.html#asynchronous-tests
    expect(new Future.error("oh no"), throwsA(equals("oh no")));

Reply:
done